Webapp/sources/services/file_attachments.py

```diff
@@ -0,0 +1,130 @@
+        .first()
+    )
+
+
+def get_blob(file_uuid: str = None) -> BlobClient:
+    """
+    Returns a binary large object (blob) from Azure Storage using the file UUID and container name.
+
+    Args:
+        file_uuid (str, optional): A string containing the UUID of the file. If not provided,
+        it is extracted from the request form data.
+
+    Returns:
+        BlobClient: An instance of the Azure Blob Client representing the specified blob.
+    """
+    # connect to azure
+    blob_service_client = connect_to_azure_blob_service()
+    # connect to blob
+    container_name = current_app.config["STORAGE_CONTAINER"]
+    if file_uuid is None:
+        file_uuid = request.form["uuid"]
+    blob_client = blob_service_client.get_blob_client(
+        container=container_name, blob=file_uuid
+    )
+    return blob_client
+
+
+def connect_to_azure_blob_service() -> BlobServiceClient:
+    """
+    Returns the Azure blob service client using the connection string from configs.
+    From this object, file attachment contents can be accessed by specifying the container name and file uuid.
+
+    Returns:
+        BlobServiceClient: An instance of the Azure Blob Service Client.
+    """
+    return BlobServiceClient.from_connection_string(
+        current_app.config["AZURE_STORAGE_CONNECTION_STRING"]
+    )
+
+
+def download_from_blob_client(blob_client: BlobClient, file_uuid: str) -> str:
+    """
+    Downloads file contents from a BlobClient and verifies integrity based on a provided file UUID.
+
+    Args:
+        blob_client: The BlobClient representing the Azure Blob Storage blob.
+        file_uuid: A string containing the UUID of the file for integrity verification.
+
+    Returns:
+        str: A base64-encoded string representing the file contents after integrity verification.
+    """
+    stream = blob_client.download_blob()
+    file_contents = stream.readall()
+    verify_integrity(file_contents, file_uuid)
+    file_attachment = base64.b64encode(file_contents).decode()
+    return file_attachment
+
+
+def verify_integrity(file_contents: bytes, file_uuid: str):
+    """
+    Verifies the integrity of file contents by comparing the SHA-256 checksum with the stored checksum
+    in the database for the file associated with the provided UUID.
+
+    Args:
+        file_contents: Bytes representing the content of the file.
+        file_uuid: A string containing the UUID of the file for integrity verification.
+
+    Raises:
+        HTTPException: Raised with a status code of 500 if the integrity check fails.
+    """
+    file_db_object = database_object_from_uuid(file_uuid)
+    if (
+        file_db_object.sha256_checksum != "not_available"
+    ):  # older files use this default value
+        checksum = services.file_attachments.sha256_from_file_contents(file_contents)
+        if file_db_object.sha256_checksum != checksum:
+            print("invalid checksum. could not verify file integrity")
+            abort(500)
```

Comment:
Feels like 500 isn't the error here, like the server hasn't errored internally, it deliberately disagrees with the request. I'd say it's a 400, but can see that opinions vary: https://stackoverflow.com/questions/35656404/whats-the-most-appropriate-http-error-code-for-a-corrupted-payload-checksum-fa

Reply:
Cheers changed it. 400 seems clearer and less ambiguous than 500 here